### PR TITLE
Bug/58403 edge browser freezes

### PR DESCRIPTION
**Pull request by jon-shipley**
* Fix issue with browser occasionally freezing under a particular set of access arrangements
* Minor upgrade angular to version 14.3.0 
* Add new runtime dependency @babel/runtime

### CODE REVIEW

File: pupil-spa/src/app/qr-code-arrival/qr-code-arrival.component.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Router } from '@angular/router'
 import { ApplicationInsightsService } from '../services/app-insights/app-insights.service';
 
 @Component({
-  selector: 'qr-code-arrival',
+  selector: 'app-qr-code-arrival',
```
Comment (GuyHarwood) [inline pupil-spa/src/app/qr-code-arrival/qr-code-arrival.component.ts:6]: why is this being changed?

Comment (jon-shipley) [reply pupil-spa/src/app/qr-code-arrival/qr-code-arrival.component.ts:6]: Lint rules : all selectors must start with 'app'.

Comment (GuyHarwood) [reply pupil-spa/src/app/qr-code-arrival/qr-code-arrival.component.ts:6]: thanks for the heads up 👍🏼

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**